infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironment.java

```diff
@@ -102,20 +103,8 @@ protected KubernetesEnvironment(
     this.secrets = secrets;
     this.configMaps = configMaps;
     this.podData = new HashMap<>();
-    pods.entrySet()
-        .forEach(
-            e -> {
-              Pod pod = e.getValue();
-              podData.put(e.getKey(), new PodData(pod.getSpec(), pod.getMetadata()));
-            });
-    deployments
-        .entrySet()
-        .forEach(
-            e -> {
-              PodTemplateSpec podTemplate = e.getValue().getSpec().getTemplate();
-              podData.put(
-                  e.getKey(), new PodData(podTemplate.getSpec(), podTemplate.getMetadata()));
-            });
+    pods.forEach((name, pod) -> podData.put(name, new PodData(pod)));
+    deployments.forEach((name, deployment) -> podData.put(name, new PodData(deployment)));
```

Comment:
I'm embarassed I didn't do it this way initially :)

Reply:
Do not worry ;)